packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1310,7 +1310,12 @@ export function returnJsonArray(jsonData: IDataObject | IDataObject[]): INodeExe
 	}
 
 	jsonData.forEach((data: IDataObject) => {
-		returnData.push({ json: data });
+		if (data.json) {
```

Comment:
Good idea. I would only type the arg instead of asserting:

```ts
jsonData.forEach((data: IDataObject & { json?: IDataObject }) => {
	if (data.json) {
		// We already have the JSON key so avoid double wrapping
		returnData.push({ ...data, json: data.json });
	} else {
		returnData.push({ json: data });
	}
});
```

Reply:
Good idea, thanks @ivov